### PR TITLE
Docs - split docs on feature changes on each full release

### PR DESCRIPTION
**Pull request by stijnmoreels**
* All the docs are marked with a version since when it's available
* All the docs are split onto changes for each full release
* The "currenly available, but not yet released" features have been split into the `preview` folder
* The `./docs/features` folder contains now the available features of the most recent version (inclusive the preview releases)

Relates to https://github.com/arcus-azure/arcus/issues/75

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -20,6 +20,18 @@ PM > Install-Package Arcus.EventGrid.All
 - Support for deserializing official Azure or custom events according to Event Grid & CloudEvent schema ([docs](features/deserializing-events))
 - Support for creating your own events ([docs](features/create-custom-events))
 
+## Older Versions
+
+- [v1.0](/v1.0/index.md)
```
Comment (tomkerkhove) [inline docs/index.md:25]: Maybe always use the x.x.x notation?

File: docs/preview/index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+title: "Home"
+layout: default
+permalink: /
```
Comment (tomkerkhove) [inline docs/preview/index.md:4]: Let's remove all permalinks & redirects

Comment (stijnmoreels) [reply docs/preview/index.md:4]: This is the preview, so, will be placed to the root index after the release. I think it's ok here. Removed the links and redirects in the older versions though.

File: docs/index.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/index.md]: Yes, done.

**Pull request merged**